### PR TITLE
docs: Create gitlab-continuous-integration.md

### PR DESCRIPTION
**Pull request by puzzledbytheweb**
## Description
Adds a small recipe with a minimal example on how to setup gitlab ci/cd for a gatsby project

### CODE REVIEW

File: docs/docs/recipes/gitlab-continuous-integration.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+build:
+  stage: build
+  script:
+    - yarn
+    - yarn build
+```
+
+3. `git push <you-remote-gitlab-repo>`
+4. Check out your pipeline working!
```
Comment (Ekwuno) [inline docs/docs/recipes/gitlab-continuous-integration.md:37]:
```suggestion
4. Check out your pipeline under the CI/CD option. 
```

File: docs/docs/recipes/gitlab-continuous-integration.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+### Additional resources
+
+- See how you can develop this simple file into something more real world [Gitlab CI/CD Docs](https://docs.gitlab.com/ee/ci/README.html)
+- Check this especially to learn how to make your newly build available for a next job - [Gitlab Job Artifacts Docs](https://docs.gitlab.com/ee/ci/pipelines/job_artifacts.html)
```
Comment (Ekwuno) [inline docs/docs/recipes/gitlab-continuous-integration.md:42]:
```suggestion
- Check this especially to learn how to make your newly build available for a next job - [Gitlab Job Artifacts Docs](https://docs.gitlab.com/ee/ci/pipelines/job_artifacts.html)

- [Getting started with GitLab CI/CD](https://gitlab.com/help/ci/quick_start/README)
```

File: docs/docs/recipes/gitlab-continuous-integration.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+---
+title: "Recipes: Continuous Integration on Gitlab"
+tableOfContentsDepth: 1
+---
+
+Push your code to gitlab and automate your production build!
+
+### Prerequisites
+
+- A [Gatsby site](/docs/quick-start)
```
Comment (Ekwuno) [inline docs/docs/recipes/gitlab-continuous-integration.md:10]: We prefer to minimise the use of external links in doc to keep the reader focused on subject matter. Can we replay this step with just having the CLI installed,  then have setting up a Gatsby site as step 1 in directions. 
```suggestion
- Make sure you have the [Gatsby CLI](docs/gatsby-cli) installed
```

File: docs/docs/recipes/gitlab-continuous-integration.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+- A [Gitlab](https://gitlab.com/) account 
+
+### Directions
+
+1. Stop your development server if it is running (`Ctrl + C` on your command line in most cases)
```
Comment (Ekwuno) [inline docs/docs/recipes/gitlab-continuous-integration.md:15]: Can we add this as a note after setting up a Gatsby site steps? see https://www.gatsbyjs.org/docs/recipes/working-with-themes#directions for reference.

Comment (puzzledbytheweb) [reply docs/docs/recipes/gitlab-continuous-integration.md:15]: Sure. Will update in a bit!

Comment (puzzledbytheweb) [reply docs/docs/recipes/gitlab-continuous-integration.md]: Included these steps in this commit - https://github.com/gatsbyjs/gatsby/pull/23367/commits/6b92831f7ca91b44be1c1db76cac482dadf40683

File: docs/docs/recipes/gitlab-continuous-integration.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    - node_modules/
+
+stages:
+  - build
+
+build:
+  stage: build
+  script:
+    - yarn
+    - yarn build
```
Comment (Ekwuno) [inline docs/docs/recipes/gitlab-continuous-integration.md:42]: Using this configuration returns a YAML invalid error, I added a few parameters to make sure the demo will build. 
```suggestion
    image: node:latest

    stages:
      - build

    cache:
      paths:
        - node_modules/

    install_dependencies:
      stage: build
      script:
        - npm install
      artifacts:
        paths:
          - node_modules/
```

File: docs/docs/recipes/gitlab-continuous-integration.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
+---
+title: "Recipes: Continuous Integration on Gitlab"
+tableOfContentsDepth: 1
+---
+
+Push your code to gitlab and automate your production build!
+
+### Prerequisites
+
+- Make sure you have the [Gatsby CLI](https://www.gatsbyjs.org/docs/gatsby-cli) installed
```
Comment (Ekwuno) [inline docs/docs/recipes/gitlab-continuous-integration.md:10]: We use relative links internally per the [style guide](https://www.gatsbyjs.org/contributing/gatsby-style-guide/)
```suggestion
- Make sure you have the [Gatsby CLI](/docs/gatsby-cli) installed
```

File: docs/docs/recipes/gitlab-continuous-integration.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
+---
+title: "Recipes: Continuous Integration on Gitlab"
+tableOfContentsDepth: 1
+---
+
+Push your code to gitlab and automate your production build!
```
Comment (Ekwuno) [inline docs/docs/recipes/gitlab-continuous-integration.md:6]: Can we get a bit of an intro to this? See the intro of https://www.gatsbyjs.org/docs/recipes/working-with-themes#creating-a-new-site-using-a-theme for context. 
```suggestion
Push your code to GitLab and automate your production build!
```

**Comment by muescha** [comment]
there is still a recipe for gitlab?

https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-recipes/recipes/gitlab-ci-cd.mdx

**Pull request merged**